### PR TITLE
If PipCount < 0, then use max / -PipCount

### PR DESCRIPTION
**Pull request by atlimit8**
Applies to:
- `AmmoPool` : `Ammo / -PipCount`
- `Cargo` : `MaxWeight / -PipCount`
- `Harvester` : `Capacity / -PipCount`
- `StoresResources` : `Capacity / -PipCount`

### CODE REVIEW

**Comment by GraionDilach** [comment]
RA2 had PipWrap: http://modenc.renegadeprojects.com/PipWrap

File: OpenRA.Mods.Common/Traits/Cargo.cs [hunk 1]
```diff
@@ -264,7 +265,7 @@ public IEnumerable<PipType> GetPips(Actor self)
 
 		PipType GetPipAt(int i)
 		{
-			var n = i * Info.MaxWeight / Info.PipCount;
+			var n = Info.PipCount >= 0 ? i * Info.MaxWeight / Info.PipCount : i / -Info.PipCount;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Cargo.cs]: This behaves counter-intuitively ingame.  If you have a half-full transport with `PipCount: -2` then I would expect that the pip row would be half full, not full.

File: OpenRA.Mods.Common/Traits/Harvester.cs [hunk 2]
```diff
@@ -420,7 +421,7 @@ public void OnNotifyResourceClaimLost(Actor self, ResourceClaim claim, Actor cla
 
 		PipType GetPipAt(int i)
 		{
-			var n = i * Info.Capacity / Info.PipCount;
+			var n = Info.PipCount >= 0 ? i * Info.Capacity / Info.PipCount : i / -Info.PipCount;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Harvester.cs]: Likewise here.  Having a harvester that continues to harvest after its pip bar is full is confusing.

**Comment by pchote** [comment]
I agree with @penev92 that this is a bit confusing, especially with the behaviour I noted above.  What's the motivation behind doing this?

**Comment by atlimit8** [comment]
`-PipCount` units per pip so that the number of pips scales with `Ammo`, `MaxWeight`, or `Capacity`.
If that is too confusing, I can roll this back to `-1` means same amount as `Ammo`, `MaxWeight`, or `Capacity`.

**Comment by Phrohdoh** [comment]
Would it make sense to turn the construct of pips entirely into an interface that has a `string Type` so implementors could do the math and always return a positive (or `0`) value?

**Comment by atlimit8** [comment]
I have a branch started with:
```c#
	public struct PipType
	{
		public readonly string Sequence;
		public readonly float Fill = 0;
	}
```
where `Sequence` is the pip sequence to use and `(int)((sequence_frame_count - 1) * Fill)` is the sequence frame to allow partial fill pips.

**Comment by atlimit8** [comment]
Perhaps, this should be reduced to supporting just `-1`. Maybe another property should be added such as `PipCountType` from something like `enum PipCountType { Total, Weight[, Frames] }` where `Total` is current/default behavior, `Weight` means each full pip is `PipCount` units[, and `Frames` means each full pip is sequence frame count - 1 (for empty) units (in other branch)].

**Comment by chrisforbes** [comment]
Big nak to magic values, magic handling of the sign value, etc.

Use another field as a magic behavior flag / mode selector.

**Pull request closed without merging**